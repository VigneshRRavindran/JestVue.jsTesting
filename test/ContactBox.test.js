import { mount } from 'vue-test-utils'
import ContactBox from '../src/components/ContactBox'

const createContactBox = (id, name, surname, selected) =>
  mount(ContactBox, {
    propsData: { id, name, surname, selected }
  })

describe('ContactBox.test.js', () => {
  // it('fullName should be the combination of name + surname', () => {
  //   const cmp = createContactBox(0, 'John', 'Doe', false)
  //   expect(cmp.vm.fullName).toBe('John Doe')
  // })

  // it('should have a selected class when the selected prop is true', () => {
  //   const cmp = createContactBox(0, 'John', 'Doe', true)
  //   expect(cmp.classes()).toContain('selected')
  // })

  it('fullName should be the combination of name + surname', () => {
    const cmp = createContactBox(0, 'John', 'Doe', false)
    expect(cmp.element).toMatchSnapshot()
  })

  it('should have a selected class when the selected prop is true', () => {
    const cmp = createContactBox(0, 'John', 'Doe', true)
    expect(cmp.element).toMatchSnapshot()
  })

  it('should emit a contact-click with its id when the component is clicked', () => {
    const cmp = createContactBox(0, 'John', 'Doe', false)
    cmp.trigger('click')

    const payload = cmp.emitted('contact-click')[0][0]
    expect(payload).toBe(0)
  })
})
