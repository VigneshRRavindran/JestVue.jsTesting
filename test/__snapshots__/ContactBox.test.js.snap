// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactBox.test.js fullName should be the combination of name + surname 1`] = `
<div
  class=""
>
  <p>
    John, Doe
  </p>
</div>
`;

exports[`ContactBox.test.js should have a selected class when the selected prop is true 1`] = `
<div
  class="selected"
>
  <p>
    John, Doe
  </p>
</div>
`;
